sklearn/metrics/_plot/regression.py

```diff
@@ -0,0 +1,356 @@
+        See :term:`Glossary <random_state>` for details.
+
+    with_residuals : bool, default=False
+        Whether or not to display the residuals on the plot for each sample.
+
+    Attributes
+    ----------
+    line_ : matplotlib Artist
+        Diagonal curve.
+
+    residual_lines_ : matplotlib Artist or None
+        Residual lines. If `with_residuals=False`, then it is set to `None`.
+
+    scatter : matplotlib Artist
+        Scatter data points.
+
+    ax_ : matplotlib Axes
+        Axes with the different matplotlib axis.
+
+    figure_ : matplotlib Figure
+        Figure containing the scatter and lines.
+
+    Examples
+    --------
+    >>> import matplotlib.pyplot as plt  # doctest: +SKIP
+    >>> from sklearn.datasets import load_diabetes
+    >>> from sklearn.linear_model import Ridge
+    >>> from sklearn.metrics import PredictionErrorDisplay
+    >>> X, y = load_diabetes(return_X_y=True)
+    >>> ridge = Ridge().fit(X, y)
+    >>> y_pred = ridge.predict(X)
+    >>> display = PredictionErrorDisplay(y_true=y, y_pred=y_pred)
+    >>> display.plot()  # doctest: +SKIP
+    >>> plt.show()      # doctest: +SKIP
+    """
+
+    def __init__(
+        self,
+        *,
+        y_true,
+        y_pred,
+        scores=None,
+        subsample=1000,
+        random_state=None,
+        with_residuals=False,
+    ):
+        self.y_true = y_true
+        self.y_pred = y_pred
+        self.scores = scores
+        self.subsample = subsample
+        self.random_state = random_state
+        self.with_residuals = with_residuals
+
+    def plot(
+        self,
+        ax=None,
+        *,
+        scatter_kwargs=None,
+        line_kwargs=None,
+        residuals_kwargs=None,
+    ):
+        """Plot visualization.
+
+        Extra keyword arguments will be passed to matplotlib's ``plot``.
+
+        Parameters
+        ----------
+        ax : matplotlib axes, default=None
+            Axes object to plot on. If `None`, a new figure and axes is
+            created.
+
+        scatter_kwargs : dict, default=None
+            Dictionary with keywords passed to the `matplotlib.pyplot.scatter`
+            call.
+
+        line_kwargs : dict, default=None
+            Dictionary with keyword passed to the `matplotlib.pyplot.plot`
+            call to draw the diagonal line.
+
+        residuals_kwargs : dict, default=None
+            Dictionary with keyword passed to the `matplotlib.pyplot.plot`
+            call to draw the residual lines. Only taken into account when
+            `with_residuals=True`.
+
+        Returns
+        -------
+        display : :class:`~sklearn.metrics.plot.PredictionErrorDisplay`
+            Object that stores computed values.
+        """
+        check_matplotlib_support("PredictionErrorDisplay.plot")
+
+        import matplotlib.pyplot as plt
+
+        random_state = check_random_state(self.random_state)
+
+        subsample = self.subsample
```

Comment:
shouldn't the Display exclusively focus on plotting logic? I feel like anything compute-related should be done outside of the display

Reply:
True